include/swift/Driver/Job.h

```diff
@@ -265,6 +265,14 @@ class Job {
   /// Whether the job wants a list of input or output files created.
   std::vector<FilelistInfo> FilelistFileInfos;
 
+  /// Response file path
+  const char *ResponseFilePath;
+
+  /// Argument vector containing a single argument pointing to the response file
+  /// path with the '@' prefix.
+  /// The argument string must be kept alive as long as the Job is alive.
+  llvm::opt::ArgStringList ResponseFileArg;
```

Comment:
This seems like overkill. Can't we just keep a `std::string` around?

Reply:
This was created to address the following issues:

1. To use the response file, we need to pass a different argument list to `TQ->addTask` that lasts as long as the `Job` object.
2. The `Job` object is `const` shortly after its creation, so we can't modify its existing argument list when it is added to the Task Queue.

I would have preferred to create the response file `ArgStringList` directly in `addPendingJobToTaskQueue` and then populate it with the name of the response file as a `std::string`, retrieved from the `Job`. However, I could not find a way to make sure the `ArgStringList` would persist as long as necessary for the `Job` to be executed.

Reply:
I didn't respond to this at the time, so clearly I understood it then, but I still don't see why that wouldn't be a `std::string`.

Reply:
`TaskQueue::addTask` takes `ArrayRef<const char *> Args` as its second argument and not a `std::string`.

Reply:
If I use a `std::string`, its memory gets cleaned up when the `Job` goes out of scope, and this is before the task actually gets executed. The compiler then tries to execute something like this:
```
swift-UH��H��0H�}�H�u�H�U�H�}��� 
```

Reply:
Ah, the string needs to live on the `llvm::opt::ArgList` that ends up being owned by the Compilation. Okay, but a `const char *` would work then, too, right? No need for a whole vector.